.circleci/config.yml

```diff
@@ -0,0 +1,112 @@
+orbs:
+  win: circleci/windows@1.0.0
+
+version: 2.1
+
+executors:
+  golang:
+    docker:
+      - image: docker.mirror.hashicorp.services/circleci/golang:1.16
+    resource_class: medium+
+  darwin:
+    macos:
+      xcode: "12.0.0"
+
+commands:
+  install-go-run-tests-unix:
+    parameters:
+      GOOS:
+        type: string
+      GOVERSION:
+        type: string
+      HOME:
+        type: string
+        default: "~"
+    steps:
+      - checkout
+      - run: curl https://dl.google.com/go/go<< parameters.GOVERSION >>.<< parameters.GOOS >>-amd64.tar.gz | tar -C << parameters.HOME >>/ -xz
+      - run: << parameters.HOME >>/go/bin/go test ./... -coverprofile=coverage.txt  -covermode=atomic
+  install-go-run-tests-windows:
+    parameters:
+      GOVERSION:
+        type: string
+    steps:
+      - checkout
+      - run: curl https://dl.google.com/go/go<< parameters.GOVERSION >>.windows-amd64.zip --output ~/go<< parameters.GOVERSION >>.windows-amd64.zip
+      - run: unzip ~/go<< parameters.GOVERSION >>.windows-amd64.zip -d ~/
+      - run: ~/go/bin/go test ./... -coverprofile=coverage.txt  -covermode=atomic
+  build-and-persist-packer-binary:
+    parameters:
+      GOOS:
+        type: string
+      GOARCH:
+        default: "amd64"
+        type: string
+    steps:
+      - checkout
+      - run: GOOS=<< parameters.GOOS >> GOARCH=<<parameters.GOARCH>> go build -ldflags="-s -w -X github.com/hashicorp/packer/version.GitCommit=${CIRCLE_SHA1}" -o ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >> .
+      - run: zip ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>.zip ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - run: rm ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./pkg/
```

Comment:
This is very packer-specific. Plugins don't have the GitCommit commit constant, we will need to implement that.  I also think the zip should have the plugin name.